### PR TITLE
GRW-2023 / Feature / Log CSP violations

### PR DESCRIPTION
**Pull request by alebedev**
## Describe your changes

Send CSP violations to /api/csp-reports
Log those reports so we can see them in Datadog

## Justify why they are needed

Ability to detect CSP errors out of development

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by alebedev** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1307** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1307" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1307?utm_source=stack-comment).

File: apps/store/src/pages/api/csp-reports.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { NextApiRequest, NextApiResponse } from 'next'
+import logger from '@/services/logger/server'
+
+const cspReports = (req: NextApiRequest, res: NextApiResponse<void>) => {
+  try {
+    const report = JSON.parse(req.body)['csp-report']
```
Comment (robinandeer) [inline apps/store/src/pages/api/csp-reports.ts:6]: Doesn't this get called with content-type headers? In that case Next.js will automatically parse the body

File: apps/store/src/pages/api/csp-reports.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    if (report) {
+      logger.child({ csp: report }).info('csp violation')
```
Comment (robinandeer) [inline apps/store/src/pages/api/csp-reports.ts:8]: Aren't we missing logging something here that lets us identify which domain that was blocked?

Should we log info or warn?

**Review by robinandeer — APPROVED** [review]
Awesome @alebedev!

File: apps/store/src/pages/api/csp-reports.ts (hunk 2 above)
Comment (alebedev) [reply apps/store/src/pages/api/csp-reports.ts:8]: This is added when sending info to Datadog

**Pull request merged**